app/controllers/users/emails_controller.rb

```diff
@@ -83,10 +83,19 @@ def authorize_user_to_edit_email
       render_not_found
     end
 
+    def request_id
+      sp_session[:request_id]
+    end
```

Comment:
I think the idea would be that we try to pull the `request_id` from `params` if it exists, and then only after we call `StoreSpMetadataInSession#call` can we reference it from `sp_session`.

Reply:
If I understand correctly this is what we're after

https://github.com/18F/identity-idp/blob/4248eded9006976b7a88dc28e5b6a1558776ef61/app/controllers/users/email_confirmations_controller.rb#L54-L58